readium/navigator/src/main/java/org/readium/r2/navigator/Navigator.kt

```diff
@@ -104,11 +104,19 @@ interface NavigatorDelegate {
  * A navigator rendering the publication visually on-screen.
  */
 interface VisualNavigator : Navigator {
+
     /**
      * Current reading progression direction.
      */
     val readingProgression: ReadingProgression
 
+    /**
+     * Returns the [Locator] to the first content element that begins on the current screen.
+     */
+    @ExperimentalReadiumApi
+    suspend fun firstVisibleElementLocator(): Locator? =
```

Comment:
This was added to start the TTS from the currently visible page instead of the start of the publication. Using the `progression` in the `currentLocator` was not precise enough, and `firstVisibleElementLocator()` makes sure to target the first HTML element fully visible or starting on the page.